shared/utils/logging_wrappers.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include "logging.hpp"
+
+#include "base-wrapper-type.hpp"
+#include "typedefs-wrappers.hpp"
+#include "typedefs-string.hpp"
+#include "typedefs-array.hpp"
+#include "typedefs-list.hpp"
+
+template<typename T>
+struct fmt::formatter<SafePtr<T>> {
+
+// Formats the point p using the parsed format specification (presentation)
+// stored in this formatter.
+template<typename FormatContext>
+auto format(SafePtr<T> const& p, FormatContext &ctx) -> decltype(ctx.out()) {
+// ctx.out() is an output iterator to write to.
+return format_to(ctx.out(), "{}|{}", p.operator bool(), fmt::ptr(p));
```

Comment:
What is this doing? It should simply log the value of the held pointer. The boolean operator for a SafePtr is given by the validity of the held handle, so if the goal is to ensure that exists first (which should be done), it should instead be something like:
```cpp
return p.isHandleValid() ? format_to(ctx.out(), "{}", fmt::ptr(p.ptr())) : format_to(ctx.out(), "{}", "<SafePtr null handle>");
```
for example. If you wish to log the held internal pointer too, more work will be required (ex, friending SafePtr to the formatter)